pkg/aws/metadata/handler_credentials.go

```diff
@@ -20,19 +20,19 @@ import (
 	"github.com/cenkalti/backoff"
 	"github.com/gorilla/mux"
 	"github.com/rcrowley/go-metrics"
-	log "github.com/sirupsen/logrus"
 	"github.com/uswitch/kiam/pkg/aws/sts"
-	"github.com/uswitch/kiam/pkg/k8s"
 	"github.com/uswitch/kiam/pkg/server"
 	"net/http"
 	"time"
 )
 
 type credentialsHandler struct {
-	roleFinder          k8s.RoleFinder
-	credentialsProvider sts.CredentialsProvider
-	clientIP            clientIPFunc
-	policy              server.AssumeRolePolicy
+	clientIP clientIPFunc
```

Comment:
The name `clientIP` is a bit misleading in a few places. We're not passing around an ip address but rather an ip resolution function. Not massively important though.